keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1361 @@
+   Set *CapacityClaimPolicySet `json:"set,omitempty" protobuf:"bytes,1,opt,name=set"`
+
+
+   // Range defines an acceptable quantity range of consuming requests.
+   // +optional
+   // +oneOf=CapacityClaimPolicy
+   Range *CapacityClaimPolicyRange `json:"range,omitempty" protobuf:"bytes,2,opt,name=range"`
+}
+
+
+// CapacityClaimPolicySet defines a discrete set of allowable capacity values for consumption.
+type CapacityClaimPolicySet struct {
+   // Default specifies the default capacity to be used for a consumption request
+   // if no value is explicitly provided.
+   //
+   // +required
+   Default resource.Quantity `json:"default,omitempty" protobuf:"bytes,1,name=default"`
+
+
+   // Options defines a discrete set of additional valid capacity values that can be requested.
+   // +optional
+   // +listType=atomic
+   Options []resource.Quantity `json:"options,omitempty" protobuf:"bytes,2,rep,name=options"`
+}
+
+
+// CapacityClaimPolicyRange defines a valid range of consuming capacity.
+type CapacityClaimPolicyRange struct {
+   // Minimum specifies the minimum capacity required for a consumption request.
+   // This field is required and serves as the default capacity to be consumed.
+   //
+   // +required
+   Minimum resource.Quantity `json:"minimum,omitempty" protobuf:"bytes,1,name=minimum"`
+
+
+   // Maximum specifies the maximum capacity that can be requested in a consumption request.
+   //
+   // +optional
+   Maximum *resource.Quantity `json:"maximum,omitempty" protobuf:"bytes,2,opt,name=maximum"`
+
+
+   // Step defines the incremental block size by which capacity can increase from the minimum.
+   //
+   // +optional
+   Step *resource.Quantity `json:"step,omitempty" protobuf:"bytes,3,opt,name=step"`
+}
+
+```
+
+#### ResourceClaimSpec's DeviceRequest
+
+```go
+
+type DeviceRequest struct {
+...
+   // Capacities define resource requirements against each capacity.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   Capacities *CapacityRequirements `json:"capacities,omitempty" protobuf:"bytes,9,opt,name=capacities"`
+}
+
+
+// CapacityRequirements defines the capacity requirements for a specific device request.
+type CapacityRequirements struct {
+   // Requests specifies the requested quantity of device capacities for a device request.
+   // +optional
+   Requests map[QualifiedName]resource.Quantity `json:"requests,omitempty" protobuf:"bytes,1,rep,name=requests"`
+
+
+   // Potentially enhancement field.
+   // Limits define the maximum amount of per-device resources allowed.
+   // This enables burstable usage when applicable.
+}
+
+
+// DeviceConstraint must have exactly one field set besides Requests.
+type DeviceConstraint struct {
+	...
+   // DistinctAttribute requires that all devices in question have this
+   // attribute and that its type and value are unique across those devices.
+   //
+   // For example, specify attribute name to get virtual devices from distinct shared physical devices.
```

Comment:
```suggestion
```

We can't refer to "name" here because there is no guarantee that devices have such an attribute.